cli/azd/pkg/azdext/extension_host.go

```diff
@@ -123,6 +146,28 @@ func (er *ExtensionHost) Run(ctx context.Context) error {
 		serviceManagers = append(serviceManagers, manager)
 	}
 
+	for _, reg := range er.frameworkServices {
+		if reg.Provider == nil {
+			return fmt.Errorf("framework service provider for language '%s' is nil", reg.Language)
+		}
+
+		manager := er.newFrameworkServiceManager(er.client)
+		if err := manager.Register(ctx, reg.Provider, reg.Language); err != nil {
+			_ = manager.Close()
+
+			for _, registered := range frameworkManagers {
+				_ = registered.Close()
+			}
+			for _, registered := range serviceManagers {
+				_ = registered.Close()
+			}
```

Comment:
Feels like we should be able to wrap some of this into a `Close()` function or similar and reuse it in other cases where registration fails.